src/components/main/flight/FlightInfoScreen.js

```diff
@@ -125,112 +122,229 @@ export default class FlightInfoScreen extends React.Component {
-    console.log(a);
+    let latArr = (this.state.arrLat * Math.PI) / 180;
+    let latDep = (this.state.depLat * Math.PI) / 180;
+    let a =
+      Math.sin(latDiff / 2) * Math.sin(latDiff / 2) +
+      Math.sin(longDiff / 2) *
+        Math.sin(longDiff / 2) *
+        Math.cos(latArr) *
+        Math.cos(latDep);
+    // console.log(a);
 
     let c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
-    console.log(c);
+    // console.log(c);
     let d = earthRad * c;
     d = d.toFixed(2);
-    console.log(d)
+    // console.log(d);
     this.setState({
       distanceTraveled: d,
     });
   }
 
+  //Calculate emissions using distance and seat class
+  calcEmissions() {
+    let dist = this.state.distanceTraveled;
+    let seat = this.state.distanceTraveled;
+    // console.log(seat);
+    //Short Flight
+    if (dist < 500) {
+      dist *= CONSTANTS.CARBON_MULTIPLIERS.short;
+    }
+    //Medium Flight
+    else if (dist < 1000) {
+      //Economy seat
+      if (this.state.seatIndex == "Economy") {
+        dist *= CONSTANTS.CARBON_MULTIPLIERS.medium_economy;
+      }
+      //Business seat
+      else if (
+        this.state.seatIndex == "Business" ||
+        this.state.seatIndex == "First Class"
+      ) {
+        dist *= CONSTANTS.CARBON_MULTIPLIERS.medium_business;
+      }
+    }
+    //Long Flight
+    else {
+      //Economy Seat
+      if (this.state.seatIndex == "Economy") {
+        dist *= CONSTANTS.CARBON_MULTIPLIERS.long_economy;
+        // console.log("Economy Long");
+        // console.log(dist);
+      }
+      //Business Seat
+      else if (this.state.seatIndex == "Business") {
+        dist *= CONSTANTS.CARBON_MULTIPLIERS.long_business;
+        // console.log("Business Long");
+        // console.log(dist);
+      }
+      //First Class Seat
+      else if (this.state.seatIndex == "First Class") {
+        dist *= CONSTANTS.CARBON_MULTIPLIERS.long_first;
+        // console.log("First Class Long");
+        // console.log(dist);
+      }
+    }
+    var emissions = Math.round(dist);
+    emissions /= 1000;
+    return emissions;
+  }
+
+  handleFlightClass(clss) {
+    this.setState({
+      seatIndex: clss,
+    });
+  }
+
+  handleTabStyle(clss) {
+    let style =
+      this.state.seatIndex == clss
+        ? styles.tabButtonActive
+        : styles.tabButtonInActive;
+    return style;
+  }
+
+  handleTabText(clss) {
+    let style =
+      this.state.seatIndex == clss
+        ? styles.tabTextActive
+        : styles.tabTextInActive;
+    return style;
+  }
+
+  handleTripIndex() {
+    if (this.state.tripIndex) {
+      this.setState({ tripIndex: false });
+      this.setState({ distanceTraveled: this.state.distanceTraveled / 2 });
+    } else {
+      this.setState({ tripIndex: true });
+      this.setState({ distanceTraveled: this.state.distanceTraveled * 2 });
```

Comment:
instead of constantly changing the actual state of distance Traveled, keep distance as oneway and in the render() check:
if(true) then display distance * 2
if(false) then display distance